kubesae/utils.py

```diff
@@ -0,0 +1,90 @@
+import invoke
+import json
+import re
+from kubesae.info import print_ansible_vars
+
+ANSIBLE_HEADER = re.compile(r"^.*\s=>\s")
+BASE_BACKUP_BUCKET = "caktus-hosting-services-backups"
+
+
+def result_to_json(result: invoke.Result):
+    value = re.sub(ANSIBLE_HEADER, "", result.stdout.strip())
+    try:
+        return json.loads(value)
+    except json.JSONDecodeError as e:
+        print(f"Something went wrong. Expected ansible header. Got {result.stdout.strip()[21]}")
+
+
+@invoke.task(name="get_db_backup")
+def get_backup_from_hosting(c, latest="daily", profile="caktus", backup_name=None, list=False):
+    """Downloads a backup from the caktus hosting services bucket
+
+    Params:
+        c (invoke.Context): the running context
+        latest (str, optional): Gets the latest backup from the specified temporal period. 
+            Defaults to "daily". Options are "daily", "weekly", "monthly", "yearly"
+        profile (str, optional): The AWS profile to allow access to the s3 bucket. DEFAULT: "caktus"
+        backup_name(str, optional): A specific backup filename.
+        list(bool, optional): If set, will list the contents of the bucket for the projects folder and exit.
+    
+    Usage:
+        $ inv utils.get-db-backup
+            Will copy the latest daily backup to project project root
+        
+        $ inv utils.get-db-backup --latest=monthly
+            Will copy the latest monthly backup to the project root
+
+        $ inv utils.get-db-backup --backup-name=yearly-2021.pgdump
+            Will copy the backup file with the name "yearly-2021.pgdump" to the project root
+        
+        $ inv utils.get-db-backup --list
+            Will list all of the backup files in the bucket for the project.
+
+        $ inv utils.get-db-backup --list --profile="client-aws"
+            Will list all of the backup files using the a locally configured AWS_PROFILE named "client-aws"
+    """
+    c.config.env = "production"
+    valid_periods = ['daily', 'weekly', 'monthly', 'yearly']
+
+    if "hosting_services_backup_folder" not in c.config.keys():
+        print("A hosting services backup folder has not been defined in tasks.py for this project.")
+        print(f"Here are a list of the currently defined backup folders:")
+        c.run(
+            f"aws s3 ls s3://{BASE_BACKUP_BUCKET}/ --profile {profile}"
```

Comment:
What AWS profile should be used for this? The one that I have currently labelled `caktus` is for the main Caktus account, not for the hosting services account, so I think I'm seeing the wrong thing.

Reply:
You are right. This should be caktus-hosting

Reply:
Well...after testing this for me, this needed to be caktus to be able to retrieve the file.

Reply:
I think this may change with role modifications, but I'm a little unsure of the exact correct way to proceed with this one. Maybe it's documentation.

Reply:
Ahh, OK, there is some magic going on that I still don't understand. If I just do `aws s3 ls` using my caktus profile, I see buckets in the main caktus account, but if I do `aws s3 ls s3://caktus-hosting-services-backups`, I see the backup folders. So your code works perfectly! (Sorry about the detour my comment caused)